src/components/element-editor/validation/validation.tsx

```diff
@@ -112,13 +100,16 @@ export const Validation = () => {
         };
 
         const updatedValues = [...validationValues, validationChild];
-        setValue("validation", updatedValues);
+        setValue("validation", updatedValues, { shouldDirty: true });
     };
 
     const handleDelete = (index: number) => {
         const currentValues = [...validationValues];
         const updatedValues = currentValues.filter((_, i) => i !== index);
-        setValue("validation", updatedValues);
+        setValue("validation", updatedValues, { shouldDirty: true });
+        setTimeout(() => {
+            setValue("validation", updatedValues, { shouldDirty: true });
+        });
```

Comment:
add this comment above the first setValue so other devs wont be confused

```
/** * shouldDirty will only dirty the field; the dirty state is not propagated to the form level * workaround is to wait for RHF to register the change and set the value again * reference: https://github.com/orgs/react-hook-form/discussions/9913#discussioncomment-4936301 */
```

Reply:
updated in latest changes